dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -680,7 +680,44 @@ public Set<WorkflowVersion> requestDOIForWorkflowVersion(@ApiParam(hidden = true
         checkNotNullEntry(result);
         PublicStateManager.getInstance().handleIndexUpdate(result, StateManagerMode.UPDATE);
         return result.getWorkflowVersions();
+    }
 
+    @PUT
+    @RolesAllowed({"curator", "admin"})
+    @Timed
+    @UnitOfWork
+    @Beta
+    @Path("/{workflowId}/requestAutomaticDOI/{workflowVersionId}")
+    @Operation(operationId = "requestAutomaticDOIForWorkflowVersion", description = "Request an automatic DOI for this version of a workflow.", security = @SecurityRequirement(name = JWT_SECURITY_DEFINITION_NAME))
+    public WorkflowVersion requestAutomaticDOIForWorkflowVersion(@Parameter(hidden = true, name = "user") @Auth User user,
+        @PathParam("workflowId") Long workflowId,
+        @PathParam("workflowVersionId") Long workflowVersionId,
+        String emptyBody) {
```

Comment:
Think this was copied from above but dropped the swagger annotation `@ApiParam(value = "This is here to appease Swagger. It requires PUT methods to have a body, even if it is empty. Please leave it empty.") String emptyBody) {` leading to the odd requestBody below. 

Would be nice to document with the openapi 3 equivalent, hopefully it makes its way to the openapi.yaml below.

Reply:
I added the equivalent openapi 3 annotation, per some examples in `UserResource`.